File: jdk5.0_src/j2se/src/solaris/native/sun/nio/ch/SocketDispatcher.c
/*
 * @(#)SocketDispatcher.c	1.7 03/12/19
 *
 * Copyright 2004 Sun Microsystems, Inc. All rights reserved.
 * SUN PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

#include "jni.h"
#include "jni_util.h"
#include "jvm.h"
#include "jlong.h"
#include "sun_nio_ch_FileDispatcher.h"

/* this is a fake c file to make the build happy since there is no
   real SocketDispatcher.c file on Solaris but there is on windows. */

static jfieldID fd_fdID;	/* for jint 'fd' in java.io.FileDescriptor */

File: jdk5.0_src/j2se/src/solaris/classes/sun/nio/ch/SocketDispatcher.java
/*
 * @(#)SocketDispatcher.java	1.9 03/12/19
 *
 * Copyright 2004 Sun Microsystems, Inc. All rights reserved.
 * SUN PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

package sun.nio.ch;

import java.io.*;

/**
 * Allows different platforms to call different native methods
 * for read and write operations.
 */

class SocketDispatcher extends NativeDispatcher
{

    int read(FileDescriptor fd, long address, int len) throws IOException {
        return FileDispatcher.read0(fd, address, len);
    }

    long readv(FileDescriptor fd, long address, int len) throws IOException {
        return FileDispatcher.readv0(fd, address, len);
    }

    int write(FileDescriptor fd, long address, int len) throws IOException {
        return FileDispatcher.write0(fd, address, len);
    }

    long writev(FileDescriptor fd, long address, int len) throws IOException {
        return FileDispatcher.writev0(fd, address, len);
    }

    void close(FileDescriptor fd) throws IOException {
        FileDispatcher.close0(fd);
    }
}
